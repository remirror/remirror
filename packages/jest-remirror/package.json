{
  "name": "jest-remirror",
  "version": "3.0.0-beta.6",
  "description": "A text editor for react built with prosemirror",
  "keywords": [
    "prosemirror",
    "react",
    "text editor"
  ],
  "homepage": "https://github.com/remirror/remirror/tree/HEAD/packages/jest-remirror",
  "repository": {
    "type": "git",
    "url": "https://github.com/remirror/remirror.git",
    "directory": "packages/jest-remirror"
  },
  "license": "MIT",
  "contributors": [
    "Ifiok Jr. <ifiokotung@gmail.com>"
  ],
  "type": "module",
  "exports": {
    ".": {
      "types": "./dist-types/index.d.ts",
      "import": "./dist/jest-remirror.js",
      "require": "./dist/jest-remirror.cjs"
    },
    "./environment": {
      "types": "./dist-types/environment.d.ts",
      "import": "./environment/dist/jest-remirror-environment.js",
      "require": "./environment/dist/jest-remirror-environment.cjs"
    },
    "./package.json": "./package.json"
  },
  "main": "./dist/jest-remirror.cjs",
  "module": "./dist/jest-remirror.js",
  "types": "./dist-types/index.d.ts",
  "files": [
    "dist",
    "dist-types",
    "environment"
  ],
  "scripts": {
    "build": "remirror-cli build"
  },
  "dependencies": {
    "@babel/runtime": "^7.22.3",
    "@remirror/core": "3.0.0-beta.6",
    "@remirror/dom": "3.0.0-beta.6",
    "@remirror/pm": "3.0.0-beta.4",
    "@remirror/preset-core": "3.0.0-beta.6",
    "@testing-library/dom": "^9.3.0",
    "@types/sanitize-html": "^2.9.0",
    "jest-prosemirror": "3.0.0-beta.4",
    "sanitize-html": "^2.10.0",
    "tiny-warning": "^1.0.3"
  },
  "devDependencies": {
    "@remirror/cli": "1.1.0-beta.0",
    "@types/node": "^16.18.34",
    "jest": "^29.5.0"
  },
  "peerDependencies": {
    "@types/node": "*",
    "jest": ">= 24.9.0"
  },
  "peerDependenciesMeta": {
    "@types/node": {
      "optional": true
    },
    "jest": {
      "optional": true
    }
  },
  "publishConfig": {
    "access": "public",
    "exports": {
      ".": {
        "types": "./dist/jest-remirror.d.ts",
        "import": "./dist/jest-remirror.js",
        "require": "./dist/jest-remirror.cjs"
      },
      "./environment": {
        "types": "./environment/dist/jest-remirror-environment.d.ts",
        "import": "./environment/dist/jest-remirror-environment.js",
        "require": "./environment/dist/jest-remirror-environment.cjs"
      },
      "./package.json": "./package.json"
    },
    "main": "./dist/jest-remirror.cjs",
    "module": "./dist/jest-remirror.js",
    "type": "module",
    "types": "./dist/jest-remirror.d.ts"
  },
  "@remirror": {
    "tsconfigs": {
      "src": {
        "compilerOptions": {
          "types": [
            "node",
            "jest"
          ]
        }
      }
    }
  },
  "preconstruct": {
    "entrypoints": [
      "index.ts",
      "environment.ts"
    ]
  }
}
