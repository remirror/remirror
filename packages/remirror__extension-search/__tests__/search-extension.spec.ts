import { extensionValidityTest, renderEditor } from 'jest-remirror';

import { SearchExtension, SearchOptions } from '../';

extensionValidityTest(SearchExtension);

function create(options?: SearchOptions) {
  const {
    add,
    nodes: { p, doc },
    view,
    commands,
  } = renderEditor<SearchExtension>([new SearchExtension(options)]);
  const node = doc(p('welcome'), p('friend'), p('hello'), p('welcome friend'));

  return { add, p, doc, view, commands, node };
}

describe('commands', () => {
  it('#find', () => {
    const { view, add, commands, node } = create();
    add(node);

    commands.search('welcome');
    expect(view.dom).toMatchSnapshot();

    commands.searchNext();
    expect(view.dom).toMatchSnapshot();

    commands.searchPrevious();
    expect(view.dom).toMatchSnapshot();
  });

  it('#replace', () => {
    const { view, add, commands, node } = create();
    add(node);

    commands.search('we');
    commands.replaceSearchResult('me');
    expect(view.dom).toMatchSnapshot();
  });

  it('#findNext, #replace', () => {
    const { view, add, commands, node } = create();
    add(node);

    commands.search('e');
    commands.searchNext();
    commands.replaceSearchResult('E');
    expect(view.dom).toHaveTextContent('welcomE');

    commands.replaceSearchResult('E');
    expect(view.dom).toHaveTextContent('friEnd');

    commands.search('hello');
    commands.replaceSearchResult('goodbye');
    expect(view.dom).toHaveTextContent('goodbye');
    expect(view.dom).toMatchSnapshot();

    commands.replaceSearchResult('abcd');
    expect(view.dom).not.toHaveTextContent('abcd');
  });

  it('#replaceAll', () => {
    const { view, add, commands, node } = create();
    add(node);

    commands.search('friend');
    expect(commands.replaceAllSearchResults.enabled('')).toBeTrue();

    commands.replaceAllSearchResults('enemy');
    expect(view.dom).toMatchSnapshot();

    expect(commands.replaceAllSearchResults.enabled('')).toBeFalse();
  });

  it('#clearSearch', () => {
    const { view, add, commands, node } = create();
    add(node);

    commands.search('friend');
    commands.clearSearch();
    expect(view.dom).toMatchSnapshot();
  });
});
