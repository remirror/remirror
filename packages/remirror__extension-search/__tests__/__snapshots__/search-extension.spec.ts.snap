// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commands #clearSearch 1`] = `
<div
  aria-label=""
  aria-multiline="true"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
  translate="no"
>
  <p>
    welcome
  </p>
  <p>
    friend
  </p>
  <p>
    hello
  </p>
  <p>
    welcome friend
  </p>
</div>
`;

exports[`commands #find 1`] = `
<div
  aria-label=""
  aria-multiline="true"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
  translate="no"
>
  <p>
    <span
      class="search highlighted-search"
    >
      welcome
    </span>
  </p>
  <p>
    friend
  </p>
  <p>
    hello
  </p>
  <p>
    <span
      class="search"
    >
      welcome
    </span>
     friend
  </p>
</div>
`;

exports[`commands #find 2`] = `
<div
  aria-label=""
  aria-multiline="true"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
  translate="no"
>
  <p>
    <span
      class="search"
    >
      welcome
    </span>
  </p>
  <p>
    friend
  </p>
  <p>
    hello
  </p>
  <p>
    <span
      class="search highlighted-search"
    >
      welcome
    </span>
     friend
  </p>
</div>
`;

exports[`commands #find 3`] = `
<div
  aria-label=""
  aria-multiline="true"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
  translate="no"
>
  <p>
    <span
      class="search highlighted-search"
    >
      welcome
    </span>
  </p>
  <p>
    friend
  </p>
  <p>
    hello
  </p>
  <p>
    <span
      class="search"
    >
      welcome
    </span>
     friend
  </p>
</div>
`;

exports[`commands #findNext, #replace 1`] = `
<div
  aria-label=""
  aria-multiline="true"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
  translate="no"
>
  <p>
    welcomE
  </p>
  <p>
    friEnd
  </p>
  <p>
    goodbye
  </p>
  <p>
    welcome friend
  </p>
</div>
`;

exports[`commands #replace 1`] = `
<div
  aria-label=""
  aria-multiline="true"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
  translate="no"
>
  <p>
    melcome
  </p>
  <p>
    friend
  </p>
  <p>
    hello
  </p>
  <p>
    <span
      class="search"
    >
      we
    </span>
    lcome friend
  </p>
</div>
`;

exports[`commands #replaceAll 1`] = `
<div
  aria-label=""
  aria-multiline="true"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
  translate="no"
>
  <p>
    welcome
  </p>
  <p>
    enemy
  </p>
  <p>
    hello
  </p>
  <p>
    welcome enemy
  </p>
</div>
`;
